src/test/scala/com/amazon/deequ/analyzers/runners/AnalyzerContextTest.scala

```diff
@@ -86,13 +86,13 @@ class AnalyzerContextTest extends AnyWordSpec
               |{"entity":"Column","instance":"item","name":"Distinctness","value":1.0},
               |{"entity":"Column","instance":"att1","name":"Completeness","value":1.0},
               |{"entity":"Multicolumn","instance":"att1,att2","name":"Uniqueness","value":0.25},
+              |{"entity":"Dataset","instance":"*","name":"Size (where: att2 == 'd')","value":1.0},
+              |{"entity":"Dataset","instance":"*","name":"Size","value":4.0},
```

Comment:
Is this change required? It is not clear from the changes that the order of the fields in the serialization process is changing. If it is not, can we remove this diff to make the PR clearer ?

Reply:
Yes.  This is required. This is the reason a test was failing.

Reply:
Thanks for the information. Is it possible to understand why it is required? The reason I ask is because of backwards compatibility. Will a serialized object that has the older order fail to get deserialized because of this change?

Reply:
I need to dig deeper to explain why it happens. My guess is that there is some kind of HashMap, HashSet that changes order of response.

Reply:
Yes. That's what happens. Below are screenshots from 2 different versions (breakpoint is on this line https://github.com/awslabs/deequ/blob/f53283e05ca93a16f054d4783e000082699fe0c3/src/main/scala/com/amazon/deequ/analyzers/runners/AnalysisRunner.scala#L201)

As you can see after map union it gives two different results. And code here (https://github.com/awslabs/deequ/blob/f53283e05ca93a16f054d4783e000082699fe0c3/src/main/scala/com/amazon/deequ/analyzers/runners/AnalyzerContext.scala#L83) iterates through the HashMap. 


Master branch
<img width="1782" alt="master" src="https://github.com/awslabs/deequ/assets/12905143/85af1c36-83d8-4ad9-903e-89713bc233b1">
feature/histogram-aggregate branch
<img width="1779" alt="feature:histogram-aggregate" src="https://github.com/awslabs/deequ/assets/12905143/199df4a3-3ec0-448e-977a-b7e3e52f0e88">